calc.go

```diff
@@ -19219,3 +19220,274 @@ func (fn *formulaFuncs) DISPIMG(argsList *list.List) formulaArg {
+
+		// Calculate dimensions of by_array
+		tempList := list.New()
+		tempList.PushBack(currentArg.Value)
+		rmin, rmax := calcColsRowsMinMax(false, tempList)
+		cmin, cmax := calcColsRowsMinMax(true, tempList)
+		key.cols, key.rows = cmax-cmin+1, rmax-rmin+1
+
+		// Validate dimensions match
+		if key.rows != res.rows {
+			msg := fmt.Sprintf("by_array dimensions (%d rows) do not match array dimensions (%d rows)",
+				key.rows, res.rows)
+			errArg := newErrorFormulaArg(formulaErrorVALUE, msg)
+			return res, &errArg
+		}
+
+		// Parse optional sort_order (default to ascending)
+		key.ascending = true
+		currentArg = currentArg.Next()
+
+		// Check if next argument exists and could be sort_order
+		if currentArg != nil {
+			// Try to parse as number
+			sortOrderArg := currentArg.Value.(formulaArg).ToNumber()
+
+			// Determine if this should be a sort_order based on argument count and position
+			expectedSortOrder := false
+			if keyCount == 0 && (argsLen == 3 || argsLen == 5 || argsLen == 7) {
+				expectedSortOrder = true
+			} else if keyCount == 1 && (argsLen == 5 || argsLen == 7) {
+				expectedSortOrder = true
+			} else if keyCount == 2 && argsLen == 7 {
+				expectedSortOrder = true
+			}
+
+			if sortOrderArg.Type == ArgError {
+				// Not a valid number
+				if expectedSortOrder {
+					// We expected a sort_order but got an error
+					return res, &sortOrderArg
+				}
+				// Otherwise, this might be the next by_array, don't consume it
+			} else {
+				// Valid number - check if it's 1 or -1
+				if sortOrderArg.Number == -1 {
+					key.ascending = false
+					currentArg = currentArg.Next()
+				} else if sortOrderArg.Number == 1 {
+					key.ascending = true
+					currentArg = currentArg.Next()
+				} else {
+					msg := fmt.Sprintf("sort_order must be 1 or -1, received %v", sortOrderArg.Number)
+					errArg := newErrorFormulaArg(formulaErrorVALUE, msg)
+					return res, &errArg
+				}
+			}
+		}
+
+		res.sortKeys = append(res.sortKeys, key)
+		keyCount++
+	}
+
+	return res, nil
+}
+
+// compareRowsForSortby compares two rows using multiple sort keys.
+// Returns true if row i should come before row j.
+func compareRowsForSortby(i, j rowWithKeys, sortKeys []sortbyKey) bool {
+	// Compare using each sort key in order
+	for keyIdx, sortKey := range sortKeys {
+		iKeys := i.sortKeys[keyIdx]
+		jKeys := j.sortKeys[keyIdx]
+
+		// Compare row keys column by column
+		for colIdx := 0; colIdx < len(iKeys) && colIdx < len(jKeys); colIdx++ {
+			comparison := compareSortbyValues(iKeys[colIdx], jKeys[colIdx])
+
+			if comparison == 0 {
+				// Equal, continue to next column
+				continue
+			}
+
+			// Apply sort order
+			if sortKey.ascending {
+				return comparison < 0
+			}
+			return comparison > 0
+		}
+
+		// All columns in this key are equal, check next key
+	}
+
+	// All keys are equal, maintain original order (stable sort)
+	return false
+}
+
+// compareSortbyValues compares two formulaArg values according to Excel sort rules:
```

Comment:
Using existing [`compareFormulaArg`](https://github.com/xuri/excelize/blob/418be6db19006f981bcf7d70666359a06ce5b0e9/calc.go#L15057-L15093) function instead of introducing new `compareSortbyValues` function.